src/keyword.rs

```diff
@@ -0,0 +1,81 @@
+use buffer::Cursor;
+use token::Token;
+
+pub trait Keyword {
+    fn ident() -> &'static str;
+
+    fn display() -> &'static str;
+}
+
+impl<K: Keyword> Token for K {
+    fn peek(cursor: Cursor) -> bool {
+        if let Some((ident, _rest)) = cursor.ident() {
+            ident == K::ident()
+        } else {
+            false
+        }
+    }
+
+    fn display() -> &'static str {
+        K::display()
+    }
+}
+
+#[macro_export]
+macro_rules! custom_keyword {
+    ($ident:ident) => {
+        custom_keyword_internal!({ pub(in self) } $ident);
```

Comment:
The default is `pub(in self)` because of an older version of the `custom_keyword_internal!` macro, and I forgot to update this call, it can be safely removed. According to https://doc.rust-lang.org/rust-by-example/mod/visibility.html, empty visibility and `in self` seem to have the same effect.